include/lo2s/perf/sample/reader.hpp

```diff
@@ -220,24 +162,13 @@ class Reader : public EventReader<T>
                            "extend to prevent the kernel from doing that"
                            " or you can increase your sampling period.";
         }
-        close();
-    }
-
-public:
-    void close()
-    {
-        if (fd_ != -1)
-        {
-            ::close(fd_);
-            fd_ = -1;
-        }
     }
 
 protected:
     bool has_cct_;
 
 private:
-    int fd_ = -1;
+    PerfEventInstance ev_instance_;
```

Comment:
Assuming you renamed the class to `EventGuard`, this member variable could simply be called `event_`.